WORKFLOW.md

```diff
@@ -1,48 +1,62 @@
 
-- Ideally a team member reviews the code, and indicates approval. For external
-  contributors, this is mandatory.
+1. A team-member or an external contributor writes code in a **feature branch**, or in their own fork. The feature branch should track an **issue**, and describe its role (e.g. “fixes 341")
 
-- PR's should not be merged until the status checks pass.
+2. Immediately after the creation of the branch, create a draft pull-request to stay up to date on progress. That way, every push on will build the project and runs the tests.
 
-- If the history is messy, the PR can be squashed, otherwise it is merged.
+3. When the code is ready, mark the pull-request as ready to review, and request a code-review from a maintainer. In the case of team members, the approval of an additional team member is required. Externals require **two** maintainers to approve the change.
 
-- When the PR is merged, an action bumps the package.json and commits that
-  change to master/develop.
+4. After a different team member reviews the code and indicates approval, the PR can be merged. If the history is messy, the PR can be squashed, otherwise, it is merged.
 
-- A tag is pushed with that code.
+### Release Cycle
+
+1. On Thursdays at 12 pm CEST, the PM Lead of the week will code-freeze `master` by creating a `release/**` branch tracking `master`. Release specific changes will be done in this branch to trigger this particular release then.
+
+2. **(TEMPORAL, see #709)** After the `release/**` action has completed, we then crease a `cd/**` branch tracking `release/**`.
+
+3. The instructions with the release, how to test and what bootstrap server to use, are then shared in our social media channels.
+
+## Actions
+
+We made active use of actions to automate tasks trivial to our workflow.
+
+- **Patch Version Bump**. When a PR to `master` is merged, an action bumps the package.json by a `patch` and commits that change to `master`.
+
+- **Tag Release**. When a PR to `master` is merged, a tag is pushed specifying that feature on that version.
+
+- **Release Version Bump**. When a `release/**` branch is pushed with a new commit, an action bumps the package.json by a `minor` and commits that change to the `release` branch. Further pushes on `release` branches would then bump the version by a `patch`.
 
 ## Branches
 
-- Master is a prerelease branch - tests _must_ pass, and code _should_ be stable,
-  but we may have issues.
+- `master`: In our case, `master` is a **prerelease** branch - tests _must_ pass, and code _should_ be stable, but its _acceptable_ to have issues.
+
+- `release/**`: Every week on Thursdays at 12 pm CEST, we cut a `release/**` branch, using an internal name to identify the release.
 
-- When we want to do a real release, we cut a `release/` branch. This should
-  bump the minor or major version (todo. workout tags)
+- `hotfix/**/**`: Hotfixes to be PR's to release branches. The release branch should then be pulled back into master.
```

Comment:
we don't need a hotfix branch format - hotfixes are just branches, and PR to release.

Reply:
Fine, let's tho try and bring some standard as we are discussing in #656